src/js/tabs/setup.js

```diff
@@ -198,21 +198,70 @@ TABS.setup.initialize = function (callback) {
             arming_disable_flags_e.hide();
         }
 
+        // DISARM FLAGS
+        // We add all the arming/disarming flags available, and show/hide them if needed.
+        var prepareDisarmFlags = function() {
+
+            var disarmFlagElements = ['NO_GYRO', 
+                                      'FAILSAFE', 
+                                      'RX_FAILSAFE', 
+                                      'BAD_RX_RECOVERY', 
+                                      'BOXFAILSAFE',
+                                      'RUNAWAY_TAKEOFF',
+                                      'THROTTLE',
+                                      'ANGLE',
+                                      'BOOT_GRACE_TIME',
+                                      'NOPREARM',
+                                      'LOAD',
+                                      'CALIBRATING',
+                                      'CLI',
+                                      'CMS_MENU',
+                                      'OSD_MENU',
+                                      'BST',
+                                      'MSP',
+                                      'PARALYZE',
+                                      'GPS'
+                                     ];
+
+            if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
+                disarmFlagElements = disarmFlagElements.concat(['RESC']);
+            }
```

Comment:
The backwards compatibility support needs to be more complete. Since arming disabled flags have been in use starting with BF3.2 they have had ordering changes as well as additions. And it's looking like for 4.0 we'll be removing one (which will also result in an ordering change). The logic here will only be correct for 3.5+ and will display incorrect messages for 3.4 and earlier.

The details of the changes by version can be found here (not updated yet for 4.0).
https://github.com/betaflight/betaflight/wiki/Arming-Sequence-&-Safety#description-of-arming-prevention-flags

In the big picture we really need to move to permanent arming disabled reason ID's and use these in MSP to insulate the configurator from these complexities.

Reply:
Looking at the code I didn't see this changes... Great link, I will use it to update for all versions.

Reply:
@etracer65 Fixed the compatibility with older versions ;) Thanks for the review!!